lib/cisco_node_utils/ntp_config.rb

```diff
@@ -0,0 +1,65 @@
+# NTP Config provider class
+#
+# Jonathan Tripathy et al., September 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node')
+
+module Cisco
+  # NtpConfig - node utility class for NTP Config configuration management
+  class NtpConfig < NodeUtil
+    attr_reader :name, :source_interface
```

Comment:
You don't need an attr_reader for `:source_interface` since you are explicitly defining the function with `def source_interface` below.